libs/vkd3d/command.c

```diff
@@ -11268,7 +11270,7 @@ static bool vkd3d_rtv_and_aspects_fully_cover_resource(const struct d3d12_resour
         const struct vkd3d_view *view, VkImageAspectFlags clear_aspects)
 {
     /* Check that we're clearing all aspects. */
-    return view->format->vk_aspect_mask == clear_aspects &&
+    return resource->format->vk_aspect_mask == clear_aspects &&
```

Comment:
Is this correct? If we have a planar image like NV12, and take a view of plane 0, we should never consider that fully-covered. I would expect to see view->format still being used here.

Reply:
This is actually precisely *why* this changed - the view format would have `COLOR`, but `clear_aspects` would also be `COLOR` whereas the resource aspects would be `PLANE_0 | PLANE_1`.

Reply:
But wouldn't the view aspects be PLANE_0 or PLANE_1 in that case? Taking a COLOR view of planar is not legal for render or sampling (without Ycbcr samplers).

Reply:
Ah ... Derp. Yes, now it makes sense.

Reply:
And to be sure, for depth-stencil, this won't cause any issues since it's not possible to view a subset of the aspects there?

Reply:
FWIW if I understand the purpose of this function correctly, we're checking here if we're discarding the *entire resource*, so even if we *could* create a depth-only view this should still be correct.

Reply:
Right.